.circleci/config.yml

```
@@ -165,113 +165,74 @@ jobs:
+    - restore_cache:
         keys:
           # when lock file changes, use increasingly general patterns to restore cache
           # https://circleci.com/docs/2.0/caching/#maven-java-and-leiningen-clojure
-          - &cache_key_beam
-            maven-repo-v1-beam-{{ .Branch }}-{{ checksum "../pom.xml" }}-{{ checksum "../ingestion-core/pom.xml" }}-{{ checksum "pom.xml" }}
-          - maven-repo-v1-beam-{{ .Branch }}-{{ checksum "../pom.xml" }}-{{ checksum "../ingestion-core/pom.xml" }}-
-          - maven-repo-v1-beam-{{ .Branch }}-{{ checksum "../pom.xml" }}-
-          - maven-repo-v1-beam-{{ .Branch }}-
-          - maven-repo-v1-beam-
-          - maven-repo-v1-
-    - &download_beam_resources
-      run:
-        name: Download resources needed for Beam tests
-        command: |
-          bin/download-schemas
-    - &maven_test
-      run:
-        name: Maven Test
-        command: |
-          mvn clean test $JAVA_COMPILER_RELEASE_FLAG
-    - &store_maven_test_results
-      store_test_results:
-        path: target/surefire-reports
-    - *report_code_coverage
-    - save_cache:
-        paths:
-          - ~/.m2
-        key: *cache_key_beam
-
-  ingestion-beam-integration:
-    <<: *ingestion_beam
-    steps:
-    # Checkout would fail if it were not the first step
-    # because checkout path would contain working dir
-    - *checkout
-    - &skip_forked_pr
-      run:
-        name: Early return if this build is from a forked PR
-        command: |
-          if [ -n "$CIRCLE_PR_NUMBER" ]; then
-            echo "Cannot pass creds to forked PRs, so marking this step successful"
-            circleci step halt
-          fi
-    - *skip_unmodified
-    - *restore_cache_beam
-    - *download_beam_resources
-    - &maven_integration_test
-      run:
-        name: Maven Integration Test
-        # Google's client libraries will check for GOOGLE_APPLICATION_CREDENTIALS
-        # and use a file in that location for credentials if present;
-        # See https://cloud.google.com/docs/authentication/production
+          - &cache_key
+            maven-repo-v1-<< parameters.module >>-{{ .Branch }}-{{ checksum "../pom.xml" }}-{{ checksum "../ingestion-core/pom.xml" }}-{{ checksum "pom.xml" }}
+          - maven-repo-v1-<< parameters.module >>-{{ .Branch }}-{{ checksum "../pom.xml" }}-{{ checksum "../ingestion-core/pom.xml" }}-
+          - maven-repo-v1-<< parameters.module >>-{{ .Branch }}-{{ checksum "../pom.xml" }}-
+          - maven-repo-v1-<< parameters.module >>-{{ .Branch }}-
+          - maven-repo-v1-<< parameters.module >>-main-
+    - when:
+        condition: << parameters.integration >>
+        steps:
+          - run:
+              name: Early return if this build is from a forked PR
+              command: |
+                if [ -n "$CIRCLE_PR_NUMBER" ]; then
+                  echo "Cannot pass creds to forked PRs, so marking this step successful"
+                  circleci step halt
+                fi
```

This `when` condition ensures that this step only appears for integration tests, but `$CIRCLE_PR_NUMBER` doesn't have an equivalent [pipeline value](https://circleci.com/docs/2.0/pipeline-variables/#pipeline-values), so the bash if statement is still necessary. 

It may be possible to use `<< pipeline.project.git_url >>`, but I dislike that it would hard-code the current url, e.g.:
```suggestion
        condition:
          and:
            - << parameters.integration >>
            - not:
                equal: [https://github.com/mozilla/gcp-ingestion, << pipeline.project.git_url >>]
        steps:
          - run:
              name: Early return for build from a forked PR
              command: |
                echo "Cannot pass creds to forked PRs, so marking this step successful"
                circleci step halt
```

---

It would be worth adding a code comment here to document this.